src/payment/strategies/affirm/affirm-payment-strategy.ts

```diff
@@ -196,87 +200,78 @@ export default class AffirmPaymentStrategy implements PaymentStrategy {
             items.push({
                 display_name: item.name,
                 sku: item.sku,
-                unit_price: item.salePrice,
+                unit_price: amountTransformer.toInteger(item.salePrice),
                 qty: item.quantity,
                 item_image_url: item.imageUrl,
                 item_url: item.url,
             });
         }
 
-        if (cart.lineItems.customItems) {
-            for (const item of cart.lineItems.customItems) {
+        for (const item of order.lineItems.giftCertificates) {
+            items.push({
+                display_name: item.name,
+                sku: '',
+                unit_price: amountTransformer.toInteger(item.amount),
+                qty: 1,
+                item_image_url: '',
+                item_url: '',
+            });
+        }
+
+        if (order.lineItems.customItems) {
+            for (const item of order.lineItems.customItems) {
                 items.push({
                     display_name: item.name,
                     sku: item.sku,
-                    unit_price: item.listPrice,
+                    unit_price: amountTransformer.toInteger(item.listPrice),
                     qty: item.quantity,
                     item_image_url: '',
                     item_url: '',
                 });
             }
         }
 
-        for (const item of cart.lineItems.giftCertificates) {
-            items.push({
-                display_name: item.name,
-                sku: '',
-                unit_price: item.amount,
-                qty: 1,
-                item_image_url: '',
-                item_url: '',
-            });
-        }
-
         return items;
     }
 
-    private _getDiscounts(): AffirmDiscount {
-        const state = this._store.getState();
-        const cart = state.cart.getCart();
+    private _getDiscounts(amountTransformer: AmountTransformer, order: Order): AffirmDiscount {
+        const discounts: AffirmDiscount = {};
 
-        if (!cart) {
-            throw new MissingDataError(MissingDataErrorType.MissingCart);
+        for (const line of order.coupons) {
+            if (line.discountedAmount > 0) {
+                discounts[line.code] = {
+                    discount_amount: amountTransformer.toInteger(line.discountedAmount),
+                    discount_display_name: line.displayName,
+                };
+            }
         }
 
-        const discounts: AffirmDiscount = {};
-        for (const line of cart.coupons) {
-            discounts[line.code] = {
-                discount_amount: line.discountedAmount,
-                discount_display_name: line.displayName,
-            };
-        }
-        for (const line of cart.discounts) {
-            discounts[line.id] = {
-                discount_amount: line.discountedAmount,
-                discount_display_name: line.id,
+        if (order.discountAmount > 0) {
+            const code = 'DISCOUNTED_AMOUNT';
+            discounts[code] = {
```

Comment:
Just `discounts.DISCOUNTED_AMOUNT` is fine.

Reply:
changed